docs/aws/audit/lambdamonitoring/rules/default_timeout_used.mdx

```diff
@@ -24,6 +24,212 @@ CBP
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to list all the Lambda functions in your AWS account. You can do this by using the AWS CLI command `aws lambda list-functions`. This command will return a JSON object with all your Lambda functions.
+
+    ```
+    aws lambda list-functions --region your-region
+    ```
+
+2. Once you have the list of all Lambda functions, you can iterate over each function to get its configuration details. You can use the `aws lambda get-function-configuration` command to get the configuration details of a specific Lambda function.
+
+    ```
+    aws lambda get-function-configuration --function-name your-function-name --region your-region
+    ```
+
+3. In the returned JSON object, look for the `Timeout` attribute. This attribute specifies the amount of time that Lambda allows a function to run before stopping it. If the `Timeout` attribute is not set or is set to a value greater than the recommended maximum, then the function is considered to have a misconfiguration.
+
+4. You can automate the above steps using a Python script. Use the `boto3` library to interact with AWS services. The script will list all Lambda functions, get their configuration details, and check the `Timeout` attribute for each function. If the `Timeout` attribute is not set or is set to a value greater than the recommended maximum, the script will print the function name and its timeout value.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: Before you start, you need to install the AWS SDK for Python (Boto3) in your environment. This can be done using pip:
+
+   ```python
+   pip install boto3
+   ```
+
+2. Establish a session with AWS: You need to establish a session with AWS using your access key, secret access key, and the region. Replace 'your_access_key', 'your_secret_access_key', and 'your_region' with your actual AWS credentials.
+
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='your_access_key',
+       aws_secret_access_key='your_secret_access_key',
+       region_name='your_region'
+   )
+   ```
+
+3. Get a list of all Lambda functions: Use the `list_functions` method to get a list of all Lambda functions in your account.
+
+   ```python
+   lambda_client = session.client('lambda')
+   functions = lambda_client.list_functions()
+   ```
+
+4. Check the timeout for each function: For each function in the list, check the 'Timeout' attribute. If it's not set, the function doesn't have a default timeout set.
+
+   ```python
+   for function in functions['Functions']:
+       if 'Timeout' not in function:
+           print(f"Lambda function {function['FunctionName']} does not have a default timeout set.")
+   ```
+
+This script will print the names of all Lambda functions that do not have a default timeout set.
+</Accordion>
+
+</AccordionGroup>
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>